mitodl/open-discussions
# Import and index MITX Online courses and content

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #3637 
Closes #3636

Related salt-ops PR: https://github.com/mitodl/salt-ops/pull/1534

#### What's this PR do?
- Imports and indexes MITX Online courses, programs, and content

#### How should this be manually tested?
Set the following in your .env file (along with the usual RC AWS credentials):
```
MITX_ONLINE_BASE_URL=https://mitxonline.mit.edu/
MITX_ONLINE_COURSES_API_URL=https://mitxonline.mit.edu/api/courses/
MITX_ONLINE_PROGRAMS_API_URL=https://mitxonline.mit.edu/api/programs/
MITX_ONLINE_LEARNING_COURSE_BUCKET_NAME=mitx-etl-mitxonline-production
```

Run the following commands:
```
manage.py backpopulate_mitxonline_data
manage.py backpopulate_mitxonline_files
```

After that you should have a little over 30 mitxonline courses (http://localhost:8063/admin/course_catalog/course/?platform=mitxonline) and 1 program.

Go to http://localhost:8063/infinite/search?o=MITx and you should see these courses show up here (along with any MITx courses hosted on edX that you may have imported previously, via `backpopulate_edx_data).

Go to http://localhost:8063/learn/search?o=MITx and you should see the same.  Click on one of the cards and the drawer should show up with details and a "Take Course" button that should link to the correct course url.

Search for text `"community-led strategies to cope with the pandemic"` (include the quotes) and you should get one search result for course "COVID-19 in Slums & Informal Settlements: Guidelines & Responses" 


#### Any background context you want to provide?
The MITX Online API will change soon to include more data, so this PR will need to be updated to reflect that.  Currently some attributes will always be empty/null because they are not returned by the API (description, price, instructors) or there are no values set (topics).

Courses from both mitxonline.mit.edu and edx.org are assigned an "Offered By" value of "mitx", but their platform values are different.

#### Screenshots (if appropriate)

![Screen Shot 2022-08-18 at 1 16 47 PM](https://user-images.githubusercontent.com/187676/185455713-b88bae30-56f0-47de-b6bb-fdd573f8f1a0.png)


![Screen Shot 2022-08-23 at 3 14 43 PM](https://user-images.githubusercontent.com/187676/186246615-8f40dbfb-c0ad-4078-ae4e-3651d98aa805.png)
